### PR TITLE
Clean up cmake variable names according to guidelines

### PR DESCRIPTION
**Pull request by HamedSabri-adsk**

### CODE REVIEW

File: cmake/test.cmake [hunk 1]
```diff
@@ -75,64 +75,64 @@ function(mayaUsd_add_test test_name)
             "used with PYTHON_MODULE or PYTHON_COMMAND")
     endif()
 
-    # set the working_dir
+    # set the WORKING_DIR
```
Comment (seando-adsk) [inline cmake/test.cmake:78]: Minor, but since this is a comment I think it should not be uppercase and be "working dir"

File: cmake/test.cmake [hunk 2]
```diff
@@ -75,64 +75,64 @@ function(mayaUsd_add_test test_name)
...
-        set(command ${PREFIX_COMMAND})
+        set(COMMAND ${PREFIX_COMMAND})
     endif()
 
-    if(pytest_code)
+    if(PYTEST_CODE)
         if(NOT PREFIX_NO_STANDALONE_INIT)
             # first, indent pycode
-            mayaUsd_indent(indented_pytest_code "${pytest_code}")
+            mayaUsd_indent(indented_PYTEST_CODE "${PYTEST_CODE}")
```
Comment (seando-adsk) [inline cmake/test.cmake:107]: Why is "indented" still lowercase?

File: cmake/test.cmake [hunk 3]
```diff
@@ -75,64 +75,64 @@ function(mayaUsd_add_test test_name)
...
-main(module=${module_name})
+import ${MODULE_NAME}
+main(module=${MODULE_NAME})
 ")
     elseif(PREFIX_PYTHON_COMMAND)
-        set(pytest_code "${PREFIX_PYTHON_COMMAND}")
+        set(PYTEST_CODE "${PREFIX_PYTHON_COMMAND}")
     else()
-        set(command ${PREFIX_COMMAND})
+        set(COMMAND ${PREFIX_COMMAND})
```
Comment (seando-adsk) [inline cmake/test.cmake:101]: I actually think that "COMMAND" is too generic for a variable name.

File: lib/mayaUsd/base/CMakeLists.txt [hunk 4]
```diff
@@ -6,27 +6,27 @@ target_sources(${PROJECT_NAME}
...
 
-set(headers
+set(HEADERS
     api.h
     debugCodes.h
 )
 
 # -----------------------------------------------------------------------------
-# promoted headers
+# promoted HEADERS
```
Comment (seando-adsk) [inline lib/mayaUsd/base/CMakeLists.txt:15]: Again comment should not be uppercase

**Comment by HamedSabri-adsk** [comment]
@seando-adsk Thanks, I will address them all today. There is also a bug introduced with this change that affects the RPATH logics which I am fixing as well.

File: cmake/test.cmake (hunk 1 above)
Comment (HamedSabri-adsk) [reply cmake/test.cmake]: @seando-adsk  please see 07d962479409f7375694679beed213da71859b0c

File: cmake/test.cmake (hunk 3 above)
Comment (HamedSabri-adsk) [reply cmake/test.cmake]: Addressed 07d962479409f7375694679beed213da71859b0c

File: lib/mayaUsd/base/CMakeLists.txt (hunk 4 above)
Comment (HamedSabri-adsk) [reply lib/mayaUsd/base/CMakeLists.txt]: Addressed 07d962479409f7375694679beed213da71859b0c

File: cmake/test.cmake (hunk 2 above)
Comment (HamedSabri-adsk) [reply cmake/test.cmake:107]: indented_PYTEST_CODE is a function argument. My goal for this PR was to tackle the set variables.

File: cmake/utils.cmake [hunk 5]
```diff
@@ -92,8 +92,8 @@ function(mayaUsd_init_rpath rpathRef origin)
     endif()
     # mayaUsd_add_rpath uses REALPATH, so we must make sure we always
     # do so here too, to get the right relative path
-    get_filename_component(origin "${origin}" REALPATH)
-    set(${rpathRef} "${origin}" PARENT_SCOPE)
+    get_filename_component(ORIGIN "${ORIGIN}" REALPATH)
+    set(${rpathRef} "${ORIGIN}" PARENT_SCOPE)
 endfunction()
```
Comment (HamedSabri-adsk) [inline cmake/utils.cmake:97]: This is a good example of why lower/upper case matters. :)

File: cmake/test.cmake (hunk 2 above)
Comment (seando-adsk) [reply cmake/test.cmake:107]: Okay, but you changed the "pytest_code" to uppercase so now it looks weird being part lower and part upper.

**Pull request merged**